### PR TITLE
Add secret upload api

### PR DESCRIPTION
**Pull request by minz1027**
- Add upload secret API endpoint
- Add README.md

Feature: https://github.com/screwdriver-cd/screwdriver/issues/148

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,12 +62,13 @@ All the possible environment variables are [defined here](config/custom-environm
...
 
-This API comes preloaded with 3 (three) resources:
+This API comes preloaded with 4 (four) resources:
  - [builds](plugins/builds/README.md)
  - [jobs](plugins/jobs/README.md)
  - [pipelines](plugins/pipelines/README.md)
+ - [secrets](plugins/secrets/README.md)
 
-An (authentication/authorization)(plugins/login/README.md) plugin.
+An (authentication/authorization)(plugins/auth/README.md) plugin.
```
Comment (stjohnjohnson) [inline README.md]: Readme is gone, please remove the link here.

File: plugins/secrets/create.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            }
+        },
+        handler: (request, reply) => {
+            const userFactory = request.server.app.userFactory;
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const secretFactory = request.server.app.secretFactory;
+            const username = request.auth.credentials.username;
+
+            return Promise.all([
+                pipelineFactory.get(request.payload.pipelineId),
```
Comment (stjohnjohnson) [inline plugins/secrets/create.js]: What if the pipeline doesn't exist?  What happens here?

File: plugins/secrets/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+const createRoute = require('./create');
+
+/**
+ * Secrets API Plugin
+ * @method register
+ * @param  {Hapi}     server            Hapi Server
+ * @param  {Object}   options           Configuration
+ * @param  {Object}   options.datastore Datastore Model
```
Comment (stjohnjohnson) [inline plugins/secrets/index.js]: Unused

File: plugins/secrets/create.js (hunk 2 above)
Comment (d2lam) [inline plugins/secrets/create.js]: @stjohnjohnson  should be addressed  now

**Comment by stjohnjohnson** [comment]
Looks like a good start to saving secrets.  👍  
secret

**Comment by tkyi** [comment]
👍 Looks good
giphy

**Pull request merged**